tests/ui/type-alias-impl-trait/defining-scopes-syntactic.rs

```diff
@@ -0,0 +1,20 @@
+#![feature(type_alias_impl_trait)]
+
+// Check that we cannot syntactically mention a TAIT without
+// it also being part of the type.
+
+type DropType<T> = ();
```

Comment:
TBH this doesn't really make sense as a test, since `type DropType<T> = ();` does not compile.

Other interesting-ish types to check would be projections I think (`<Foo as Id>::Out`, `<() as Map<Foo>>::Out`).

Reply:
I wanted to keep it as a sanity check